cli/foundation-models/system/inference/automatic-speech-recognition/asr-batch-endpoint.sh

```diff
@@ -0,0 +1,96 @@
+set -x
+# the commands in this file map to steps in this notebook: https://aka.ms/azureml-infer-batch-sdk-asr
+
+# script inputs
+registry_name="azureml"
+subscription_id="<SUBSCRIPTION_ID>"
+resource_group_name="<RESOURCE_GROUP>"
+workspace_name="<WORKSPACE_NAME>"
+
+# This is the model from system registry that needs to be deployed
+model_name="openai-whisper-large"
+
+# Validate the existence of the model in the registry and get the latest version
+model_list=$(az ml model list --name ${model_name} --registry-name ${registry_name} 2>&1)
+if [[ ${model_list} == *"[]"* ]]; then
+    echo "Model doesn't exist in registry. Check the model list and try again."; exit 1;
+fi
+version_temp=${model_list#*\"version\": \"}
+model_version=${version_temp%%\"*}
+
+version=$(date +%s)
+endpoint_name="asr-$version"
+job_name="asr-job-$version"
+
+# todo: fetch compute_sku from the min_inference_sku tag of the model
+compute_sku="Standard_DS5_v2"
```

Comment:
I'm not sure what will happen if the workflows run all these notebooks/bash scripts in parallel. Each notebook/bash will try to create or update the compute cluster. If the request is for the same settings - same sku, min/max nodes, it would probably be fine, as it would be a no-op update. If the request is to change the SKU/settings for a cluster, I don't know what happens, especially if other jobs are already queued.

It might be better to set a min/max instance count and SKU type that would be compatible for all of these models, then use that in all the scripts/notebooks.  Thinking of when the notebook and bash script workflows run, how fast do we need the jobs to complete? If all 8 notebooks and all 8 bash scripts were to run at the same time, each requesting 1 node, and the cluster can scale to max of 3, how long will it take to complete all 16 jobs? Is that acceptable or not?

Reply:
Would it be easier to have unique names for the compute clusters across the scripts? If we have enough quota, it shouldn't cause any issues.

Reply:
I would still leverage the job queueing and node scalability of one cluster, but knowing how many nodes would be required if all jobs run at once would be useful. IDK if it is only batch jobs running on this cluster or other teams use the same cluster.

Reply:
The lines, both in CLI and SDK notebook examples, where the compute is deleted could cause us issues. If these notebooks run at different times using the same cluster and then all the notebooks try to delete the cluster while it still has jobs running or queued, that will create problems. Perhaps we can put the lines to delete the cluster as comments so it won't actually run in a workflow?

Reply:
The commands for creating clusters would also cause issues, right? Wouldn't it give us an error if we try to create a compute cluster that already exists?